PCbuild/regen.vcxproj

```diff
@@ -0,0 +1,180 @@
+    <Import Project="pyproject.props" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros" />
+  <ItemDefinitionGroup>
+    <ClCompile>
+      <PreprocessorDefinitions>_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+    </ClCompile>
+    <Link>
+      <SubSystem>Console</SubSystem>
+    </Link>
+  </ItemDefinitionGroup>
+  <ItemGroup>
+    <ClCompile Include="..\Programs\_freeze_importlib.c" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="pythoncore.vcxproj">
+      <Project>{cf7ac3d1-e2df-41d2-bea6-1e2556cdea26}</Project>
+      <Private>true</Private>
+      <ReferenceOutputAssembly>true</ReferenceOutputAssembly>
+      <CopyLocalSatelliteAssemblies>false</CopyLocalSatelliteAssemblies>
+      <LinkLibraryDependencies>true</LinkLibraryDependencies>
+      <UseLibraryDependencyInputs>false</UseLibraryDependencyInputs>
+    </ProjectReference>
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="..\Grammar\Grammar"></None>
+    <None Include="..\Grammar\Tokens"></None>
+ 
+    <None Include="..\Include\graminit.h"></None>
+    <None Include="..\Python\graminit.c"></None>
+    <None Include="..\Include\token.h"></None>
+    <None Include="..\Include\opcode.h"></None>
+
+    <None Include="..\Include\Python-ast.h"></None>
+    <None Include="..\Python\Python-ast.c"></None>
+
+    <None Include="..\Lib\keyword.py"></None>
+    <None Include="..\Lib\symbol.py"></None>
+
+    <None Include="..\Doc\library\token-list.inc"></None>
+
+  </ItemGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+  <ImportGroup Label="ExtensionTargets">
+
+  </ImportGroup>
+  <Target Name="_RegenGrammar" BeforeTargets="Build">
+    <!-- Regenerate Include/graminit.h and Python/graminit.c from Grammar/Grammar using pgen-->
+    <Exec Command="&quot;$(OutDir)python$(PyDebugExt).exe&quot; -m Parser.pgen &quot;..\Grammar\Grammar&quot; &quot;..\Grammar\Tokens&quot; &quot;..\Include\graminit.h.new&quot; &quot;..\Python\graminit.c.new&quot;" />
+    <Copy SourceFiles="..\Include\graminit.h.new" DestinationFiles="..\Include\graminit.h">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedH" />
+    </Copy>
+    <Copy SourceFiles="..\Python\graminit.c.new" DestinationFiles="..\Python\graminit.c">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedC" />
+    </Copy>
+    <Warning Text="Grammar updated. You will need to rebuild pythoncore to see the changes." Condition="'@(_UpdatedH)' != '' and '@(_UpdatedC)' != '' and $(Configuration) == 'Debug'" />
+  </Target>
+  <Target Name="_RegenAST_H" AfterTargets="_RegenGrammar">
+    <!-- Regenerate Include/Python-ast.h using Parser/asdl_c.py -h -->
+    <Exec Command="&quot;$(OutDir)python$(PyDebugExt).exe&quot; &quot;..\Parser\asdl_c.py&quot; -h &quot;..\Include\Python-ast.h.new&quot; &quot;..\Parser\Python.asdl&quot;" />
+    <Copy SourceFiles="..\Include\Python-ast.h.new" DestinationFiles="..\Include\Python-ast.h">
```

Comment:
This is going to leave a mess in the source directory. Can you put the temporary files in `$(IntDir)` and copy from there?

Reply:
Sure, the `Makefile` just puts them in the source directory, I was copying that behaviour

Reply:
Yeah, the Makefile doesn't care :) We do a better job on Windows